network/quic/dialer.go

```diff
@@ -0,0 +1,37 @@
+package quic
+
+import (
+	"crypto/tls"
+	"time"
+
+	h "github.com/ConsenSys/handel"
+	quic "github.com/lucas-clemente/quic-go"
+)
+
+//Dialer is an interface responsible for creating session between two peers
+type Dialer interface {
+	//startDial create seession between two peers, this is blocking method
+	startDial(identity h.Identity, out chan *result)
+}
+
+type quicDialer struct {
+	handshakeTimeout time.Duration
+}
+
+func newQuicDialer(handshakeTimeout time.Duration) Dialer {
+	return &quicDialer{handshakeTimeout}
+}
+
+func (q quicDialer) startDial(identity h.Identity, out chan *result) {
+	tlsCfg := &tls.Config{InsecureSkipVerify: true}
```

Comment:
I think you should be have two separate dialers implementation where one uses this insecure option and the other one does not, and the rest of the logic is the same.